salt/modules/nilrt_ip.py

```diff
@@ -0,0 +1,515 @@
+# -*- coding: utf-8 -*-
+'''
+The networking module for NI Linux Real-Time distro
+
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import logging
+import time
+
+
+log = logging.getLogger(__name__)
+
+# Import salt libs
+import salt.utils
+import salt.utils.validate.net
+import salt.exceptions
+# Import third party libs
+import pyconnman
+import dbus
+
+
+# Define the module's virtual name
+__virtualname__ = 'ip'
+
+SERVICE_PATH = '/net/connman/service/'
+_CONFIG_TRUE = ['yes', 'on', 'true', '1', True]
+
+
+def __virtual__():
+    '''
+    Confine this module to NI Linux Real-Time based distros
+    '''
+    if __grains__['os_family'] == 'NILinuxRT':
+        if _get_state() == 'offline':
```

Comment:
Because `_get_state()` can raise an exception, we need to catch that here and then handle it appropriately instead of letting it bubble up.